webapp/graphite/finders/standard.py

```diff
@@ -66,9 +75,21 @@ def find_nodes(self, query):
             else:
               for datasource_name in RRDReader.get_datasources(absolute_path):
                 if match_entries([datasource_name], datasource_pattern):
+                  if found_leafs >= settings.METRICS_FIND_WHISPER_TRUNCATE_THRESHOLD:
+                    truncated_leafs += 1
+                    continue
+                  found_leafs += 1
                   reader = RRDReader(absolute_path, datasource_name)
                   yield LeafNode(metric_path + "." + datasource_name, reader)
 
+    total_leafs = found_leafs + truncated_leafs
+    if total_leafs >= settings.METRICS_FIND_WHISPER_FAILURE_THRESHOLD:
+      raise Exception("Failed metrics find due to METRICS_FIND_WHISPER_FAILURE_THRESHOLD reached (%d > %d) for '%s'" %
+                      (total_leafs, settings.METRICS_FIND_WHISPER_FAILURE_THRESHOLD, clean_pattern))
+    elif truncated_leafs > 0:
```

Comment:
As discussed, refocussed on global threshold for now. Will work later on whisper finder